eurito_daps/packages/utils/openaire_utils.py

```diff
@@ -0,0 +1,165 @@
+    '''
+    #iterate through records
+    for record in records:
+
+        #create object
+        recordObj = get_record_object(record, outputType)
+
+        #if software, find related EC projects and create relationship with related ECprojects via association table
+        if outputType == "software":
+            recordObj = link_record_with_project(record, recordObj)
+
+        #add object into database
+        #dbsession.add(recordObj) #this gives an error, try incorporating this:
+        local_object = dbsession.merge(recordObj)
+        dbsession.add(local_object)
+
+        dbsession.commit()
+
+def get_record_object(cur_record, outputType):
+    '''A utility function, which returns a ORM-type object according to the specified output type
+
+    Args:
+        record (dict): contains metadata of a record (e.g. record of type "software" would store fields such as "title", "pid", "creator", etc.)
+        outputType (str): type of record to be extracted from OpenAIRE API. Accepts "software", "datasets", "publications", "ECProjects"
+    '''
+    if outputType == 'software':
+        return openaire_orm.SoftwareRecord(title=cur_record['title'], pid=cur_record['pid'], creators=str(cur_record['creators']) )
+    if outputType == 'ECProjects':
+        return openaire_orm.ECProjectRecord(title=cur_record['title'], projectcode=cur_record['projectcode'])
+
+
+def parse_software_soup_rt (cur_soup):
+    '''A utility function, which parses software records from XML and returns a list of records with software that are related to EC Projects
+
+    Args:
+        cur_soup (XML string): contains string formatted in XML, that was obtained from BeautifulSoup request to the API
+    '''
+    outputlist = list()
+    results = cur_soup.find_all(re.compile("^oaf:result"))
+    for result in results:
+
+        #check if related to EC projects
+        projectcodes = result.find_all('code')
+        #print(projectcodes)
+        #if code tag exists, then it is related to EC project
+        if projectcodes:
+            out_obj = dict()
+            out_obj['projectcodes'] = projectcodes
+
+            pid = result.find('pid') #doi identifier
+            #print(pid.text)
+            out_obj['pid'] = pid.text
+
+            title = result.find('title')
+            #print(title.text)
+            out_obj['title'] = title.text
+
+            creators = result.find_all('creator')
+            #print(creators)
+            out_obj['creators'] = creators
+
+            outputlist.append(out_obj)
+    return outputlist
+
+def parse_projects_soup_rt (cur_soup):
+    '''A utility function, which parses EC project records from XML, returns a list of records
+
+    Args:
+        cur_soup (XML string): contains string formatted in XML, that was obtained from BeautifulSoup request to the API
+    '''
+    outputlist = list()
+    results = cur_soup.find_all(re.compile("^oaf:project"))
+    for result in results:

```

Comment:
I think you could/should replace the contents of this function with 3 lines and maintain readability:

```
results = cur_soup.find_all(re.compile("^oaf:project"))
return [{'title': r.find('title').text, 
         'project_code': r.find('code').text}
         for r in results]

```